### PR TITLE
Do not expose username and password in values.yaml for grafana.

### PR DESCRIPTION
**Pull request by gyliu513**
/cc @douglas-reid @sdake @linsun 

Fixed https://github.com/istio/istio/issues/7476

### CODE REVIEW

**Comment by gyliu513** [comment]
@douglas-reid @sdake @linsun @costinm any comments for this? With this PR, we will not need to expose user name and password for grafana.

File: install/kubernetes/helm/istio/values.yaml [hunk 1]
```diff
@@ -401,8 +401,9 @@ grafana:
   storageClassName: ""
   security:
     enabled: false
-    adminUser: admin
-    adminPassword: admin
+    secretName: grafana
+    usernnameKey: username
+    passworkdKey: password
```
Comment (douglas-reid) [inline install/kubernetes/helm/istio/values.yaml]: There are multiple typos here. Have you tested and verified this actually works?

Comment (gyliu513) [reply install/kubernetes/helm/istio/values.yaml]: Done, thanks @douglas-reid

File: install/kubernetes/helm/istio/README.md [hunk 2]
```diff
@@ -56,7 +56,43 @@ $ helm init --service-account tiller
...
+
+Encode password, you can chage the password to the password as you want:
+```
+$ echo -n '1f2d1e2e67df' | base64
+MWYyZDFlMmU2N2Rm
+```
+
+Set the namespace where Istio was installed:
+```
+$ NAMESPACE=istio-sysem
```
Comment (morvencao) [inline install/kubernetes/helm/istio/README.md]: `istio-sysem` should be `istio-system`

Comment (gyliu513) [reply install/kubernetes/helm/istio/README.md]: Done

**Comment by gyliu513** [comment]
The Kiali PR at https://github.com/istio/istio/pull/7967#pullrequestreview-147021943 got approved by @douglas-reid , seems we agree on this approach.

@douglas-reid @costinm can you help check if we can merge this as well?

File: install/kubernetes/helm/istio/charts/grafana/templates/deployment.yaml [hunk 3]
```diff
@@ -46,13 +46,13 @@ spec:
...
-                key: username
+                name: {{ .Values.security.secretName }}
+                key: {{ .Values.security.usernameKey }}
           - name: GF_SECURITY_ADMIN_PASSWORD
             valueFrom:
               secretKeyRef:
-                name: grafana
-                key: password
+                name: {{ .Values.security.secretName }}
+                key: {{ .Values.security.passwordKey }}
```
[the remark this answers is not in the document]
Comment (gyliu513) [reply install/kubernetes/helm/istio/charts/grafana/templates/deployment.yaml]: Done

File: install/kubernetes/helm/istio/values.yaml [hunk 4]
```diff
@@ -404,8 +404,9 @@ grafana:
   storageClassName: ""
   security:
     enabled: false
-    adminUser: admin
-    adminPassword: admin
+    secretName: grafana
+    usernameKey: username
```
[the remark this answers is not in the document]
Comment (gyliu513) [reply install/kubernetes/helm/istio/values.yaml]: This was only enabled when customer enabled `grafana.security` @douglas-reid

File: install/kubernetes/helm/istio/values.yaml [hunk 5]
```diff
@@ -404,8 +404,9 @@ grafana:
...
+    passwordKey: password
```
[the remark this answers is not in the document]
Comment (gyliu513) [reply install/kubernetes/helm/istio/values.yaml]: Done

File: install/kubernetes/helm/istio/values.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (sdake) [reply install/kubernetes/helm/istio/values.yaml]: Could probably still place these keys under grafana with your more recent changes.  We should default to secure by default imo.

[the remark this answers is not in the document]
Comment (gyliu513) [reply install/kubernetes/helm/istio/values.yaml]: @douglas-reid what do you think?

@sdake shall we handle this in another PR?

**Comment by sdake** [comment]
@gyliu513 not sure if a 1.0.6 is planned, but if so, could you backport your fix to release-1.0?

Cheers
-steve

**Comment by gyliu513** [comment]
@sdake Sure, let me create PRs for this.

**Pull request merged**